### ISSUE TITLE
[SUGGESTION]:  Real Evo Dex Filter

### ISSUE THREAD

**Issue by DragonGirl218** [pokeclicker/pokeclicker#4882]
### Current Behavior

Currently, if you are doing the Real Evolution Challenge and evolve a Pokemon, leaving you without the base Pokemon, the missing base Pokemon appears as Caught in the Pokedex. While this makes sense given that the Pokemon has been caught before, it doesn't make the missing Pokemon stand out like the Uncaught filter does.

### Improved Behavior

I think it would be useful to give the Pokedex a filter exclusively to the Real Evo Challenge for Pokemon that are Caught but not in the party.

### Reason for change

As an example: when the Halloween event was running, I caught a Spooky Bulbasaur and immediately moved to Johto. It eventually evolved to Spooky Venusaur, leaving me without Spooky Bulbasaur or Ivysaur. I went to check my Pokedex filtered to "Uncaught" and it did not show Spooky Bulbasaur or Ivysaur. It was not until I went back to Kanto and saw the map highlight for routes with Pokemon I don't have that I realized I needed to catch another two Spooky Bulbasaurs to have the full evolution tree.

On the surface one might say that the Pokedex filter isn't needed if the route will still show the highlight. But I also have a Spooky Togepi and Togetic but not Spooky Togekiss because my Real Evo file isn't in Sinnoh yet. Once I get to Sinnoh I'll evolve them, and that will leave me without a Spooky Togepi after the event has already ended. I've been told I can get the Togepi by breeding the Togetic, but I am forgetful and easily distracted, so if I don't do it immediately, I might forget, and the Togepi won't be noted as not being in my party if I check the Pokedex because it has technically been caught before, meaning that if I'm really dumb, I might not realize I'm missing that Togepi until next year's event. But if I can look at a "Caught but not in party" filter, it will remind me to breed the Togetic far sooner than that.

It's a minor thing, but it'd be a nice little bit of QoL for the challenge.

### Detailed Description

Basically just a Pokedex filter that cross-references Pokemon that have been caught and Pokemon that are in the party, and displays caught Pokemon that aren't in the party. Because this is basically impossible without the Real Evo challenge, it could potentially only be displayed if that challenge is active, or has ever been active on the save file.

**Comment by VideoGameWizard** [comment]
Yeah I feel like this is weird that they don't already have it like this. I would also like it if they change it.

**Comment by EatPant2nd** [comment]
But the Pokémon that go away DO show as uncaught. The issue you had was that you already had "another" Bulbasaur caught. The Pokédex only shows "base" uncaught if you don't have the ID already captured. So, say, if I have a Pinkan Rattata but no base Rattata, it will NOT show base Rattata as uncaught because I do have **a** Rattata. So the issue is less about Real Evolution and more about how the Pokedex handles "uncaught" entries.
